README.md

```diff
@@ -77,264 +81,298 @@ Community and partner projects:
+backbone.fc = torch.nn.Identity()
+
+# Build the SimCLR model.
 model = SimCLR(backbone)
 
-# lightly exposes building blocks such as loss functions
+# Prepare transform that creates multiple random views for every image.
+transform = transforms.SimCLRTransform(input_size=32, cj_prob=0.5)
+
+# Combine views from multiple images into a batch.
+collate_fn = MultiViewCollate()
+
+# Create a dataset from your image folder.
+dataset = data.LightlyDataset(input_dir="./my/cute/cats/dataset/", transform=transform)
+
+# Build a PyTorch dataloader.
+dataloader = torch.utils.data.DataLoader(
+    dataset,  # Pass the dataset to the dataloader.
+    batch_size=128,  # A large batch size helps with the learning.
+    shuffle=True,  # Shuffling is important!
+    collate_fn=collate_fn,
+)
+
+# Lightly exposes building blocks such as loss functions.
 criterion = loss.NTXentLoss(temperature=0.5)
 
-# get a PyTorch optimizer
-optimizer = torch.optim.SGD(model.parameters(), lr=1e-0, weight_decay=1e-5)
+# Get a PyTorch optimizer.
+optimizer = torch.optim.SGD(model.parameters(), lr=0.1, weight_decay=1e-6)
+
+# Train the model.
+for epoch in range(10):
+    for (view0, view1), targets, filenames in dataloader:
+        z0 = model(view0)
+        z1 = model(view1)
+        loss = criterion(z0, z1)
+        loss.backward()
+        optimizer.step()
+        optimizer.zero_grad()
+        print(f"loss: {loss.item():.5f}")
 ```
 
 You can easily use another model like SimSiam by swapping the model and the
 loss function.
+
 ```python
-# PyTorch module for the SimSiam model
-class SimSiam(nn.Module):
+# PyTorch module for the SimSiam model.
+class SimSiam(torch.nn.Module):
     def __init__(self, backbone):
         super().__init__()
         self.backbone = backbone
-        self.projection_head = SimSiamProjectionHead(512, 512, 128)
-        self.prediction_head = SimSiamPredictionHead(128, 64, 128)
+        self.projection_head = heads.SimSiamProjectionHead(512, 512, 128)
+        self.prediction_head = heads.SimSiamPredictionHead(128, 64, 128)
 
     def forward(self, x):
-        f = self.backbone(x).flatten(start_dim=1)
-        z = self.projection_head(f)
+        features = self.backbone(x).flatten(start_dim=1)
+        z = self.projection_head(features)
         p = self.prediction_head(z)
         z = z.detach()
         return z, p
 
+
 model = SimSiam(backbone)
 
-# use the SimSiam loss function
+# Use the SimSiam loss function.
 criterion = loss.NegativeCosineSimilarity()
 ```
+
 You can [find a more complete example for SimSiam here.](https://docs.lightly.ai/self-supervised-learning/examples/simsiam.html)
 
 
 Use PyTorch Lightning to train the model:
 
 ```python
-trainer = pl.Trainer(max_epochs=max_epochs, devices=1, accelerator="gpu")
-trainer.fit(
-    model,
-    dataloader
-)
+from pytorch_lightning import Trainer
+
+trainer = Trainer(max_epochs=10, devices=1, accelerator="gpu")
+trainer.fit(model, dataloader)
```

Comment:
This is a bit misleading because the model above cannot be plugged into the lightning trainer.

Reply:
I added an example model for pytorch lightning as well.